fairscale/optim/oss.py

```diff
@@ -434,14 +434,14 @@ def _broadcast_params(
             # Since all the parameters are already sorted per increasing size, we only need to consider the first ones.
             while i_bucketed < len(params) and offset + params[i_bucketed].numel() < buffer_size:
                 end = offset + params[i_bucketed].numel()
-                if rank == self_rank:
+                if global_rank == self_rank:
```

Comment:
This function might be simpler if it didn't pass group and self_rank and instead just used self. That way, the correctness would be obvious. self_rank is ambiguous as to whether it is global or local.

```suggestion
                if global_rank == self.global_rank:
```